testing/endtoend/components/eth1/depositor.go

```diff
@@ -0,0 +1,208 @@
+package eth1
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/prysmaticlabs/prysm/v3/encoding/bytesutil"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	contracts "github.com/prysmaticlabs/prysm/v3/contracts/deposit"
+	eth "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	e2e "github.com/prysmaticlabs/prysm/v3/testing/endtoend/params"
+	"github.com/prysmaticlabs/prysm/v3/testing/endtoend/types"
+	"github.com/prysmaticlabs/prysm/v3/testing/util"
+)
+
+const depositGasLimit = 4000000
+
+var gweiPerEth = big.NewInt(int64(params.BeaconConfig().GweiPerEth))
+
+func amtInGwei(deposit *eth.Deposit) *big.Int {
+	amt := big.NewInt(0).SetUint64(deposit.Data.Amount)
+	return amt.Mul(amt, gweiPerEth)
+}
+
+func computeDeposits(offset, nvals int, partial bool) ([]*eth.Deposit, error) {
```

Comment:
Would be nice to add clarity this operates over deterministic validator keys and produces a list of deposits for them with certain conditions, such as offsets, total number, and whether or not they should be partial deposits

Reply:
added more comments, ptal!